src/capture.rs

```diff
@@ -71,23 +77,59 @@ pub async fn event(
     }))
 }
 
-pub fn process_single_event(_event: &Event) -> Result<ProcessedEvent> {
-    // TODO: Put actual data in here and transform it properly
+pub fn process_single_event(event: &RawEvent, query: &EventQuery) -> Result<ProcessedEvent> {
+    let distinct_id = match &event.distinct_id {
+        Some(id) => id,
+        None => match event.properties.get("distinct_id").map(|v| v.as_str()) {
+            Some(Some(id)) => id,
+            _ => return Err(anyhow!("missing distinct_id")),
+        },
+    };
+
     Ok(ProcessedEvent {
-        uuid: Uuid::new_v4(),
-        distinct_id: Uuid::new_v4().simple().to_string(),
-        ip: String::new(),
+        uuid: event.uuid.unwrap_or_else(uuid_v7),
+        distinct_id: distinct_id.to_string(),
+        ip: query.client_ip.clone().unwrap_or_default(),
         site_url: String::new(),
         data: String::from("hallo I am some data 😊"),
-        now: String::new(),
+        now: query.now.clone().unwrap_or_default(),
         sent_at: String::new(),
-        token: String::from("tokentokentoken"),
+        token: query.token.clone().unwrap_or_default(),
     })
 }
 
+pub fn extract_and_verify_token(events: &[RawEvent]) -> Result<String, String> {
```

Comment:
```suggestion
pub fn extract_and_verify_token(events: &[RawEvent]) -> Result<String> {
```

I'd love it if we could stick to proper error types rather than `Err(<string error>)`. Right now `anyhow` is a bit of a temporary fix while we evolve the codebase, but the next steps would be sorting out some concrete error types.

The other cool things we can then do would be adding context with something like [error_context](https://docs.rs/error-context/latest/error_context/) or eyre's `Context`, more formatting, etc. We can't do that so well with a string

Reply:
Agreed to do later, probably with a `thiserror` enum